### ISSUE TITLE
Refactor and clean up the createStagingDeployIssue code

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#9999]
# Problem

Coming from https://github.com/Expensify/App/pull/9946#discussion_r924758705

The code for creating staging deploy issue is very hard to follow and leads to bugs when new features are added or code is changed as engineers have tough time following the code.

Also sometimes we can manually move PRs to be internalQA after being added to the checklist - try to check if the PRs in main section are indeed internal or not before adding them in - do not rely on previous checklist in terms of this.

# Why is this important?

Intoruduced more bugs in the issues which slows down deploy and takes more time to fix.

# Solution

We should do:
- clean up and refactor the code/flow so it is easier to follow
- make sure to add more comments which will clearly indicate what are the formats/structures of the arrays/objects and in general comments explaining the flow
- Find a nicer way to get the URL from the internal QA section than this https://github.com/Expensify/App/blob/d24ab21aa1b76791978bf76ea4682d1bfee462e0/.github/libs/GithubUtils.js#L192

And:
>Also sometimes we can manually move PRs to be internalQA after being added to the checklist - try to check if the PRs in main section are indeed internal or not before adding them in - do not rely on previous checklist in terms of this.

Lets also update naming to make sure it is not outdated: 

>  replace StagingDeployCash with DeployChecklist, and LockCashDeploys with LockDeploys

cc @roryabraham

**Comment by mountiny** [comment]
Low priority for now

**Comment by mountiny** [comment]
Focusing on API refactors and n7

**Comment by mountiny** [comment]
Still in my backlog, I will let this once to the pool if anyone is interested

**Comment by roryabraham** [comment]
Overall I think we may have lost the context for this issue and the problem is unclear. I think we should refine the problem statement or close this out.

**Comment by roryabraham** [comment]
I didn't really have anything major planned for the checklist other than removing some outdated naming (i.e: replace `StagingDeployCash` with `DeployChecklist`, and `LockCashDeploys` with `LockDeploys`)

**Comment by mountiny** [comment]
Alright, I think from the bugZero perspective, we can close this one out. The issue got created when we were trying to add a new feature to the checklist and it revealed how easy it is to miss something in the complicated code. However, we havent had issues with the checklist for a long time or needs to change it so I think we could close this now and if there would be any new bigger changes required for the checklist in future, the refactoring should take place along with it.

Let me know if you disagree.

**Comment by puneetlath** [comment]
Sounds good to me.
